src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,243 @@
+        link.setHref(ixbrlLocation);
+
+        return Arrays.asList(link);
+    }
+
+    /**
+     * Get the description values, which currently contains the period end date. This data is
+     * required for the filing description.
+     *
+     * @param periodEndDate - account's period end date
+     * @return {@link Map<String,String>} containing period end date, to match filing model
+     */
+    private Map<String, String> getDescriptionValues(LocalDate periodEndDate) {
+        Map<String, String> descriptionValues = new HashMap<>();
+        descriptionValues.put(PERIOD_END_ON, periodEndDate.toString());
+
+        return descriptionValues;
+    }
+
+    /**
+     * Get the description for the filing. The description for the account type is retrieved by
+     * using DocumentDescriptionHelper class.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param periodEndDate - account's period end date
+     * @return accounts description.
+     * @throws IOException
+     */
+    private String getFilingDescription(AccountsType accountsType, LocalDate periodEndDate)
+        throws IOException {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(PERIOD_END_ON, periodEndDate);
+
+        return documentDescriptionHelper
+            .getDescription(accountsType.getFilingDescriptionKey(), parameters);
+    }
+
+    /**
+     * Validates the ixbrl against TNEP. This validation is driven by the env. variable and it can
+     * be disable.
+     */
+    private boolean isValidIxbrl() {
+
+        boolean isIxbrlValid = true;
+        if (environmentReader.getMandatoryBoolean(DISABLE_IXBRL_VALIDATION_ENV_VAR)) {
+            //TODO TNEP validation needs to be added. Copy logic from abridged. (STORY SFA-574)
+        }
+
+        return isIxbrlValid;
+    }
+
+    /**
+     * Generates the ixbrl by calling the document generator passing the the end point and
+     * company/account id.
+     *
+     * @return The location where the service has stored the generated ixbrl.
+     */
+    private String generateIxbrl() {
+        //TODO: call the document generator's new end point to get generate ixbrl. Implementation does NOT exist yet. (STORY SFA-574)
+        return  documentGeneratorCaller.generateIxbrl(getServiceURL());
+    }
+
+    /**
+     * Build the document render service URL.
+     *
+     * @return the document render service end point e.g. "http://chs-dev:4082/document-render/store"
+     */
+    private String getServiceURL() {
+        return getMandatoryEnvVariable(DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR) +
+            DOCUMENT_RENDER_SERVICE_END_POINT;
+    }
+
+    /**
+     * Get environment variable value for the passed variable.
+     *
+     * @return environment variable value.
+     */
+    private String getMandatoryEnvVariable(String envVariable) {
+        return environmentReader.getMandatoryString(envVariable);
```

Comment:
If a static final EnvironmentReader is declared at the class level, we can declare static strings at the class level to hold environment variables. This means that we don't have to read the same environment variables every time a method is invoked. For example:

```
public class Class {

   private static final EnvironmentReader READER = new EnvironmentReaderImpl();

   private static String DOCUMENT_RENDER_SERVICE_HOST = READER.getMandatoryString("DOCUMENT_RENDER_SERVICE_HOST");

}
```